dayrepo/snippets/views.py

```diff
@@ -225,117 +226,182 @@ def get(self, request, snippet_id):
-                    start_point=req.getlist("start_point")[i + 1],
-                    end_point=req.getlist("end_point")[i + 1],
-                    via_point=req.getlist("via_point")[i],
-                    client=req.getlist("client")[i],
-                    goods=req.getlist("goods")[i],
-                    load_situation=req.getlist("load_situation")[i],
-                    load_mileage=req.getlist("load_mileage")[i],
-                    hollow_mileage=req.getlist("hollow_mileage")[i],
-                    is_load_situation=req.getlist("is_load_situation")[i],
-                )
-                if process.is_load_situation == "on":
-                    process.is_load_situation = True
-                else:
-                    process.is_load_situation = False
-                post_process[i].delete()
-                process.save()
-            post_snippet=snippet
-            # トップ画面へ
-            return redirect(to="snippet_list")
+                process.is_load_situation = False
+            post_process[i].delete()
+            process.save()
+        post_snippet = snippet
+        # トップ画面へ
+        return redirect(to="snippet_list")
+
 snippet_edit = SnippetEditView.as_view()
 
+
+class DbDeletePattern():
+    def snippet_del(self,target_id:int):
+        dutiestrouble = DutiesTrouble.objects.get(
+            snippet_id=target_id
+        )
+        dutiestrouble.delete()
+        process = get_list_or_404(
+            Process, snippet_id=target_id
+        )
+        for process_object in process:
+            process_object.delete()
+
+        snippet = Snippet.objects.get(
+            pk=target_id
+        )
+        snippet.delete()
+
+
+        checklist = Checklist.objects.get(
+            pk=snippet.checklist_id.id
+        )
+
+        # 対象のsnippetが削除されたことを反映し、
+        # 未提出日報として表示、判定させる
+        checklist.is_snippet_make = False
+        checklist.save()
+
+    # 未提出 data の checklist を削除する
+    def checklist_del(self,target_id:int):
+        checklist = Checklist.objects.get(
+        pk=target_id
+        )
+        checklist.delete()
+
+    # 提出済み data をすべて削除する
+    def all_del(self,target_id:int):
+        snippet = Snippet.objects.get(
+            pk=target_id
+        )
+        checklist = Checklist.objects.get(
+            pk=snippet.checklist_id.pk
+        )
+        dutiestrouble = DutiesTrouble.objects.get(
+            snippet_id=target_id
+        )
+        dutiestrouble.delete()
+        process_list = get_list_or_404(
+            Process, 
+            snippet_id=target_id
+        )
+        for process in process_list:
+            process.delete()
+        snippet.delete()
+        checklist.delete()
+
+def db_delete(request:HttpRequest, target_id:int, delete_type:str):
+    # 受け取ったdelete_typeから
+    # 削除対象を判別し、実行クラスの処理を選択
+    del_pattern = DbDeletePattern()
+    if request.method != 'POST':
+        raise ValueError(
+            "本来入るはずのない処理に入りました。お手数ですが、システムにお問合せください。"
+        )
```

Comment:
request.method != 'POST' に引っかかった場合 DbDeletePattern をインスタンスする意味がないのと、コメントなどを含めた一連の処理としては例外処理は別物なので、以下のようにした方がいいと思います。

```suggestion
def db_delete(request:HttpRequest, target_id:int, delete_type:str):
    if request.method != 'POST':
        raise ValueError(
            "本来入るはずのない処理に入りました。お手数ですが、システムにお問合せください。"
        )

    # 受け取ったdelete_typeから
    # 削除対象を判別し、実行クラスの処理を選択
    del_pattern = DbDeletePattern()
```